HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h

```diff
@@ -48,4 +48,47 @@ inline bool cudaCheck_(const char* file, int line, const char* cmd, cudaError_t
 
 #define cudaCheck(ARG) (cudaCheck_(__FILE__, __LINE__, #ARG, (ARG)))
 
+
+namespace {
+
+  template <typename T>
+  [[noreturn]] inline void abortOnCudaErrorVerbose(
+      const char* file, int line, const char* cmd, const char* error, const char* message, T const& description) {
+    std::ostringstream out;
+    out << "\n";
+    out << file << ", line " << line << ":\n";
+    out << "cudaCheck(" << cmd << ");\n";
+    out << error << ": " << message << "\n";
+    out << description << "\n";
+    throw std::runtime_error(out.str());
+  }
+
+}  // namespace
+
+template <typename T>
+inline bool cudaCheckVerbose_(const char* file, int line, const char* cmd, CUresult result, T const& description) {
+  if (__builtin_expect(result == CUDA_SUCCESS, true))
```

Comment:
How about `LIKELY` macro?
https://github.com/cms-patatrack/cmssw/blob/614ee0b4e0d1d457d6bbf7d51e0799f330fa242b/FWCore/Utilities/interface/Likely.h#L14

Reply:
I didn't pick it up because I am using the same file also outside of CMSSW.

Reply:
I think we should consider using `LIKELY` in the version going into CMSSW, because it is only conditionally defined as `__builtin_expect` (so on a build where `__builtin_expected` is missing for some reason the direct usage here would fail to compile).

Reply:
OK... done.
I would have preferred `BOOST_LIKELY` as something I could use outside of CMSSW, but adding a dependency in a .h file would mess up the BuildFiles, so `LIKELY` it is.